### PR TITLE
Add endogenous DNA calculation script

### PR DESCRIPTION
**Pull request by aidaanva**
This pull requests adds a small script that calculates the endogenous DNA content from a single samtools flagstat file or when performing a filtering in the bam based on mapping quality it calculates it from the pre filtering samtools flagstat and the post filtering samtools flagstat. Please ignore nf-core linting failure, @jfy133 will be fixing this in the next pull request for a template syncing.

Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

## PR checklist
 - [x] This comment contains a description of changes (with reason)
 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [x] If necessary, also make a PR on the [nf-core/eager branch on the nf-core/test-datasets repo]( https://github.com/nf-core/test-datasets/pull/new/nf-core/eager)
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
 - [ ] Make sure your code lints (`nf-core lint .`).
 - [x] Documentation in `docs` is updated
 - [x] `CHANGELOG.md` is updated
 - [x] `README.md` is updated

**Learn more about contributing:** https://github.com/nf-core/eager/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: bin/endorS.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+parser.add_argument('samtoolsfiles', metavar='<samplefile>.stats', type=str, nargs='+',
+                    help='output of samtools flagstat in a txt file (at least one required). If two files are supplied, the mapped reads of the second file is divided by the total reads in the first, since it assumes that the <samplefile.stats> are related to the same sample. Useful after BAM filtering')
+parser.add_argument('-v','--version', action='version', version='%(prog)s 0.1')
+parser.add_argument('--output', '-o', nargs='?', help='specify a file format for an output file. Options: <json> for a MultiQC json output. Default: none')
+parser.add_argument('--name', '-n', nargs='?', help='specify name for the output file. Default: extracted from the first samtools flagstat file provided')
+args = parser.parse_args()
+
+#Open the samtools flag stats pre-quality filtering:
+try:
+    with open(sys.argv[1], 'r') as pre:
```
Comment (maxibor) [inline bin/endorS.py]: Can you parse all the input files and arguments with argparse ?
Something like this:
Just switch to `parser.add_argument('samtools_pre', help="flag stats pre-quality filtering")`
Here is an [example](https://github.com/maxibor/adrsm/blob/826bcd7ec9ed271401cc5cc4c0d0a545050eb65a/adrsm#L8) with argparse.

File: bin/endorS.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    mappedPre = float((re.findall(r'([0-9]+) \+ [0-9]+ mapped ',contentsPre))[0])
+    #Calculation of endogenous DNA pre-quality filtering:
+    endogenousPre = float("{0:.2f}".format(round((mappedPre / totalReads * 100), 2)))
+except:
+    print("Incorrect input, please provide at least a samtools flag stats as input\nRun:\npython endorS.py --help \nfor more information on how to run this script")
+    sys.exit()
+#Check if the samtools stats post-quality filtering have been provided:
+try:
+    #Open the samtools flag stats post-quality filtering:
+    with open(sys.argv[2], 'r') as post:
```
Comment (maxibor) [inline bin/endorS.py]: Same as above

File: main.nf [hunk 3]
```diff
@@ -1210,7 +1210,7 @@ process samtools_flagstat {
     file(bam) from ch_mapping_for_samtools_flagstat
 
     output:
-    file "*.stats" into ch_flagstat_for_multiqc
+    tuple val(prefix), file("*stats") into ch_flagstat_for_multiqc,ch_flagstat_for_endorspy
```
Comment (maxibor) [inline main.nf]: Nice that you use the tuple !
It begs for another PR to replace all the `set` (being deprecated) by `tuple` [https://www.nextflow.io/docs/latest/process.html?highlight=tuple#input-of-type-tuple](https://www.nextflow.io/docs/latest/process.html?highlight=tuple#input-of-type-tuple)

File: main.nf [hunk 4]
```diff
@@ -1359,15 +1359,55 @@ process samtools_flagstat_after_filter {
...
+    input:
+    tuple val(name), file(stats), file(poststats) from ch_allflagstats_for_endorspy
+
+    output:
+    file "*.json" into ch_endorspy_for_multiqc
+
+    script:
+    prefix = "${name}"
+    """
+    endorS.py ${stats} ${poststats} -o json -n ${name}
```
Comment (maxibor) [inline main.nf]: You will need to revise this line after modifying the argument parsing (files go after flags)

File: bin/endorS.py (hunk 1 above)
Comment (jfy133) [reply bin/endorS.py]: @aidaanva

File: bin/endorS.py (hunk 2 above)
Comment (jfy133) [reply bin/endorS.py]: @aidaanva

File: main.nf (hunk 3 above)
Comment (jfy133) [reply main.nf]: I think that is the plan with the TSV input (as we have to update all the input/outputs anyway)

**Comment by maxibor** [comment]
Also, could you add a sentence/paragraph in the docs (in `ouput.md`, in the multiqc section) describing this output ?

**Comment by jfy133** [comment]
To cover the TODO from @maxibor 

- [ ] fix argparse in the endorS.py script 
- [ ] update nextflow process following above
- [x] convert all sets to tuples in nextflow script <- will be occuring in different PR
- [x] add output documentation

**Comment by aidaanva** [comment]
@maxibor I updated the python script, could you take a look? Does it make sense what I did or did I misunderstand what you meant?

File: main.nf (hunk 4 above)
Comment (maxibor) [reply main.nf]: Could you re-shuffle the order of the options (first), and arguments (last) @aidaanva ?

Comment (aidaanva) [reply main.nf]: @maxibor the options and the arguments can be reshuffled, @jfy133 should update the code now

**Pull request merged**